plugins/history/databasemanager.cpp

```diff
@@ -0,0 +1,123 @@
+#include "kapplication.h"
+#include "databasemanager.h"
+#include "kopetecontact.h"
+#include "kopetemessage.h"
+#include "kopeteaccount.h"
+#include "kopetechatsession.h"
+#include "kopeteprotocol.h"
+#include "kopetechatsessionmanager.h"
+
+#include "databaseconstants.h"
+
+#include <KStandardDirs>
+#include <QVariant>
+
+
+DatabaseManager* DatabaseManager::mInstance = 0;
+
+DatabaseManager::DatabaseManager(QObject *parent)
+	: QObject(parent)
+{
+	mInstance = this;
+}
+
+void DatabaseManager::bindQueryValue(QSqlQuery query, QString columnName, QString value)
+{
+	query.bindValue(columnName, value);
+}
+
+DatabaseManager::~DatabaseManager()
+{
+
+}
+
+void DatabaseManager::initDatabase(DatabaseManager::DatabaseType dbType)
+{
+	//Close the database, in case it is open.
+	if (db.isOpen()) {
+		db.close();
+	}
+
+	//Create the database tables based on the selected database system
+	switch (dbType) {
+	case SQLITE:
+		//For SQLite, we store the database in the user's application data folder.
+		QString dbPath = KStandardDirs::locateLocal("appdata", "kopete_history.db");
+		db = QSqlDatabase::addDatabase("QSQLITE", "kopete-history");
+		db.setDatabaseName(dbPath);
+
+		//Abort in case the database creation/opening fails.
+		if (!db.open()) {
+			//Database not open
+			return;
+		}
+
+		//If the messages table does not exist, lets create it.
+		db.exec(DatabaseConstants::sql_createMessagesTable());
+	}
+}
+
+void DatabaseManager::insertMessage(Kopete::Message &message)
+{
+	QSqlQuery query(db);
```

Comment:
I've added an error message in case the database is not open. I can catch this in the UI and display a warning.